test/common/router/header_formatter_speed_test.cc

```diff
@@ -40,8 +40,7 @@ static void bmEvaluateHeaders(benchmark::State& state) {
   HeaderParserPtr header_parser = HeaderParser::configure(headers_to_add);
 
   for (auto _ : state) { // NOLINT: Silences warning about dead store
-    header_parser->evaluateHeaders(*request_header, *request_header,
-                                   *Http::StaticEmptyHeaders::get().response_headers, *stream_info);
+    header_parser->evaluateHeaders(*request_header, {request_header.get()}, *stream_info);
```

Comment:
Was there any effect on the speed test?

Reply:
I think there should have no obvious impact. But I can take a check and paste result here.

Reply:
Result after this PR:
```
---------------------------------------------------------------
Benchmark                     Time             CPU   Iterations
---------------------------------------------------------------
bmEvaluateHeaders/2         316 ns          316 ns      2155852
bmEvaluateHeaders/4         982 ns          982 ns       715449
bmEvaluateHeaders/6        1564 ns         1564 ns       441221
bmEvaluateHeaders/8        2118 ns         2118 ns       336647
bmEvaluateHeaders/10       2701 ns         2701 ns       258921
bmEvaluateHeaders/12       3310 ns         3309 ns       212408
bmEvaluateHeaders/14       3943 ns         3943 ns       181631
bmEvaluateHeaders/16       4442 ns         4442 ns       156797
bmEvaluateHeaders/18       5178 ns         5178 ns       138296
bmEvaluateHeaders/20       5737 ns         5737 ns       120386
```

Reply:
Result before this PR:
```
---------------------------------------------------------------
Benchmark                     Time             CPU   Iterations
---------------------------------------------------------------
bmEvaluateHeaders/2         341 ns          341 ns      2057969
bmEvaluateHeaders/4        1032 ns         1032 ns       671125
bmEvaluateHeaders/6        1670 ns         1670 ns       416722
bmEvaluateHeaders/8        2330 ns         2329 ns       301196
bmEvaluateHeaders/10       3018 ns         3017 ns       234001
bmEvaluateHeaders/12       3717 ns         3717 ns       191802
bmEvaluateHeaders/14       4269 ns         4269 ns       165998
bmEvaluateHeaders/16       4873 ns         4873 ns       143795
bmEvaluateHeaders/18       5529 ns         5528 ns       126884
bmEvaluateHeaders/20       6219 ns         6219 ns       113101
```